src/Documents/Queries/IndexQueryBase.php

```diff
@@ -0,0 +1,167 @@
+    }
+
+    /**
+     * Maximum number of records that will be retrieved.
+     * @deprecated use LIMIT in RQL instead
+     * @return int page size
+     */
+    public function getPageSize(): int
+    {
+        return $this->pageSize;
+    }
+
+    /**
+     * Maximum number of records that will be retrieved.
+     * @deprecated use LIMIT in RQL instead
+     * @param int $pageSize Sets the value
+     */
+    public function setPageSize(int $pageSize): void
+    {
+        $this->pageSize = $pageSize;
+        $this->pageSizeSet = true;
+    }
+
+    /**
+     * When set to true server side will wait until result are non stale or until timeout
+     * @return bool true if server should wait for non stale results
+     */
+    public function isWaitForNonStaleResults(): bool
+    {
+        return $this->waitForNonStaleResults;
+    }
+
+    /**
+     * When set to true server side will wait until result are non stale or until timeout
+     * @param bool $waitForNonStaleResults Sets the valueQer
+     */
+    public function setWaitForNonStaleResults(bool $waitForNonStaleResults): void
+    {
+        $this->waitForNonStaleResults = $waitForNonStaleResults;
+    }
+
+    public function getWaitForNonStaleResultsTimeout(): Duration
+    {
+        return $this->waitForNonStaleResultsTimeout;
+    }
+
+    public function setWaitForNonStaleResultsTimeout(Duration $waitForNonStaleResultsTimeout): void
+    {
+        $this->waitForNonStaleResultsTimeout = $waitForNonStaleResultsTimeout;
+    }
+
+    public function toString(): string
+    {
+        return $$this->query;
+    }
+
+    public function equals(?object &$o): bool {
+        if ($this == $o) return true;
+        if (($o == null) || (get_class($this) != get_class($o))) return false;
+
+        /** @var IndexQueryBase $that */
+        $that = $o;
+
+        if ($this->pageSize != $that->pageSize) return false;
+        if ($this->pageSizeSet != $that->pageSizeSet) return false;
+        if ($this->start != $that->start) return false;
+        if ($this->waitForNonStaleResults != $that->waitForNonStaleResults) return false;
+        if (($this->query != null) ? (strcmp($this->query, $that->query)) : ($that->query != null)) return false;
+        return $this->waitForNonStaleResultsTimeout != null ? $this->waitForNonStaleResultsTimeout->equals($that->waitForNonStaleResultsTimeout) : $that->waitForNonStaleResultsTimeout == null;
+    }
+
+    public function hashCode(): int
+    {
+        $result = $this->pageSize;
+        $result = 31 * $result + ($this->pageSizeSet ? 1 : 0);
+        $result = 31 * $result + ($this->query != null ? HashUtils::hashCode($this->query) : 0);
+        $result = 31 * $result + $this->start;
+        $result = 31 * $result + ($this->waitForNonStaleResults ? 1 : 0);
+        $result = 31 * $result + ($this->waitForNonStaleResultsTimeout != null ? $this->waitForNonStaleResultsTimeout->hashCode() : 0);
+        return $result;
+    }
```

Comment:
usually you don't have to rewrite equals and hashcode functions. Moreover those are autogenerated in java. This is used for having such objects as map keys to compute hash backup and compare objects based on contents not via reference.

Reply:
at the beginning i didn't understood this methods so as you can see I just copied them. Now I understand it, and I will remove this methods soon